types/proofs.go

```diff
@@ -0,0 +1,24 @@
+package types
+
+// SinglePartitionProofLen represents the number of bytes in a single partition
+// PoRep or PoSt proof. The total length of a PoSt or PoRep proof equals the
+// product of SinglePartitionProofLen and the number of partitions.
+const SinglePartitionProofLen int = 192
+
+// PoStProof is the byte representation of the Proof of SpaceTime proof
+type PoStProof []byte
+
+// PoRepProof is the byte representation of the Seal Proof of Replication
+type PoRepProof []byte
+
+// ProofPartitions returns the number of partitions used to create the PoRep
+// proof.
+func (s PoRepProof) ProofPartitions() PoRepProofPartitions {
+	return NewPoRepProofPartitions(len(s) / SinglePartitionProofLen)
```

Comment:
This should really fail here if `len(s)` is not `2*SinglePartitionProofLen`. Getting a `UnknownPoStProofPartitions` that's going to produce bad numbers somewhere else will just make it hard to track bugs.

Also, using integer math, this is going to produce:

* `UnknownPoStProofPartitions` for 0 <= len(s) < 384
* `TwoPoRepProofPartitions` for 384 <= len(s) < 576
* `UnknownPoStProofPartitions`  for 576 <= len(s) < ♾

Is this what you want? Is 500 bytes a valid TwoPoRepProofPartitions?

Reply:
I really wanted the type of this function to be `(s PoRepProof) ProofPartitions() PoRepProofPartitions` and not `(s PoRepProof) ProofPartitions() (PoRepProofPartitions, error)`. 

> Is 500 bytes a valid TwoPoRepProofPartitions?

The rust-fil-proofs system will reject (with an error) a 500 byte PoRep proof. The rust-fil-proofs system will never produce a 500 byte PoRep proof, either. By relying on the rust-fil-proofs system to reject the buggy proof, I could avoid duplicating the runtime validation into go-filecoin.

That said, it's not much software to enforce the invariant on both sides of the FFI wall. Would you agree to a `panic` in the event that we can't map the PoRepProof bytes cleanly to a PoRepProofPartitions value?